### PR TITLE
fix: don't spam ledger with adapter.getAddress() calls for EVM chains

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

This PR ensures we only call `adapter.getAddress()` once in `deriveEvmAccountIdsAndMetadata` for Ledger.
The result is twofolds:

1. Ledger chain connect for EVM chains is consistently happy on initial "Connect" click (i.e without cache)
2. The connection is now much faster since we only call `adapter.getAddress()` once

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes https://github.com/shapeshift/web/issues/6188

Theoretically low, effectively very high risk given the domain touched. If there is any issue with this, accounts will be fully borked.

> What protocols, transaction types or contract interactions might be affected by this PR?

## Testing

- Ledger connection to EVM accounts is consistently happy
- After a full cache clear, accounts for other wallets are happy on wallet connection/import

### Ledger EVM accounts
 
https://github.com/shapeshift/web/assets/17035424/81276a74-effc-486e-b515-d814eb4cc264

### MM connect after a full cache clear

 
image

### Native connect after a full cache clear

image
image
image

### CODE REVIEW

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* `develop`
  * **PR #6209** <a href="https://app.graphite.dev/github/pr/shapeshift/web/6209?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/6209?utm_source=stack-comment).

File: src/lib/account/evm.ts [hunk 1]
```diff
@@ -9,13 +10,29 @@ import {
...
+        if (isEvmChainId(chainId)) {
+          const adapter = assertGetEvmChainAdapter(chainId)
+          return await adapter.getAddress({ accountNumber, wallet })
+        }
+      }
+    }
+    return null
+  })()
+    // if this throws for any reason, we don't want to make this whole async function reject
+    .catch(() => null)
```
Comment (gomesalexandre) [inline src/lib/account/evm.ts:35]: i.e do not let this bubble up as unhandled promise rejection

**Review by gomesalexandre — COMMENTED** [review]
@kaladinlight can't stamp my own PR but tested the cleanup commit and confirmed this still does what it says on the box, ty!

**Review by 0xApotheosis — APPROVED** [review]
✅ Ledger EVM connections are happy (and speedy)
✅ Non-ledger wallets work, and have the same accounts as `develop` with a fresh cache

**Review by kaladinlight — APPROVED** [review]

**Pull request merged**